modules/framework/galasa-parent/dev.galasa.framework.api.monitors/src/main/java/dev/galasa/framework/api/monitors/internal/routes/MonitorsDetailsRoute.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.framework.api.monitors.internal.routes;
+
+import static dev.galasa.framework.api.common.ServletErrorMessage.*;
+
+import java.util.regex.Matcher;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import dev.galasa.framework.api.beans.generated.GalasaMonitor;
+import dev.galasa.framework.api.common.HttpRequestContext;
+import dev.galasa.framework.api.common.InternalServletException;
+import dev.galasa.framework.api.common.MimeType;
+import dev.galasa.framework.api.common.ProtectedRoute;
+import dev.galasa.framework.api.common.QueryParameters;
+import dev.galasa.framework.api.common.ResponseBuilder;
+import dev.galasa.framework.api.common.ServletError;
+import dev.galasa.framework.api.common.ServletErrorMessage;
+import dev.galasa.framework.api.monitors.internal.IKubernetesApiClient;
+import dev.galasa.framework.api.monitors.internal.MonitorTransform;
+import dev.galasa.framework.spi.FrameworkException;
+import dev.galasa.framework.spi.rbac.RBACService;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.models.V1Deployment;
+
+public class MonitorsDetailsRoute extends ProtectedRoute {
+
+    // Regex to match /monitors/{monitorName} and /monitors/{monitorName}/
+    // where {monitorName} can consist of the following characters:
+    // - Alphanumeric characters (a-zA-Z0-9)
+    // - Underscores (_)
+    // - Dashes (-)
```

Comment:
The doc in the error messages above suggested that a '.' (dot) was allowed also ?